x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/CcrLicenseIT.java

```diff
@@ -140,60 +140,64 @@ public void onFailure(final Exception e) {
     }
 
     public void testAutoFollowCoordinatorLogsSkippingAutoFollowCoordinationWithNonCompliantLicense() throws Exception {
-        // Update the cluster state so that we have auto follow patterns and verify that we log a warning in case of incompatible license:
-        CountDownLatch latch = new CountDownLatch(1);
-        ClusterService clusterService = getInstanceFromNode(ClusterService.class);
-        clusterService.submitStateUpdateTask("test-add-auto-follow-pattern", new ClusterStateUpdateTask() {
-
-            @Override
-            public ClusterState execute(ClusterState currentState) throws Exception {
-                AutoFollowPattern autoFollowPattern = new AutoFollowPattern("test_alias", Collections.singletonList("logs-*"),
-                    null, null, null, null, null, null, null, null, null, null, null);
-                AutoFollowMetadata autoFollowMetadata = new AutoFollowMetadata(
-                    Collections.singletonMap("test_alias", autoFollowPattern),
-                    Collections.emptyMap(),
-                    Collections.emptyMap());
-
-                ClusterState.Builder newState = ClusterState.builder(currentState);
-                newState.metaData(MetaData.builder(currentState.getMetaData())
-                    .putCustom(AutoFollowMetadata.TYPE, autoFollowMetadata)
-                    .build());
-                return newState.build();
-            }
-
-            @Override
-            public void clusterStateProcessed(String source, ClusterState oldState, ClusterState newState) {
-                latch.countDown();
-            }
-
-            @Override
-            public void onFailure(String source, Exception e) {
-                latch.countDown();
-                fail("unexpected error [" + e.getMessage() + "]");
-            }
-        });
-        latch.await();
-
         final Logger logger = LogManager.getLogger(AutoFollowCoordinator.class);
         final MockLogAppender appender = new MockLogAppender();
         appender.start();
         appender.addExpectation(
-                new MockLogAppender.ExceptionSeenEventExpectation(
-                        getTestName(),
-                        logger.getName(),
-                        Level.WARN,
-                        "skipping auto-follower coordination",
-                        ElasticsearchSecurityException.class,
-                        "current license is non-compliant for [ccr]"));
-        Loggers.addAppender(logger, appender);
+            new MockLogAppender.ExceptionSeenEventExpectation(
```

Comment:
Before the auto following was bootstrapped from thread pool scheduler, 
but now auto followers for new remote clusters are bootstrapped when
a new cluster state is published.